ChainSafe/web3.unity
# Web3Auth Wallet GUI

Web3Auth Wallet GUI! 

We now have a way to show users what is actually happening with Web3Auth! Woo!

The Web3Auth login scene now has an enable wallet boolean, when enabled the wallet GUI will pop up in game & prompt users to sign transactions if specified as well as additional config options.

![image](https://github.com/ChainSafe/web3.unity/assets/57473220/12ebf435-725b-4a0c-b4a9-022a4f5d3a89)

- Devs can decide if they'd like to show or hide the wallet icon (shift + tab will need to be used if so)
- Changing the wallet logo & icon can be done by assigning it in the editor under WalletLogo/WalletIcon.
- Auto pop up wallet on tx will for the GUI to show whenever there's an incoming transaction.
- Auto transaction preference.
- Ability to see private key similar to Metamask.
- Ability to customize background, menu colours, font, primary and secondary text colours.
- As everything scales relatively for all devices, menu positioning can easily be customized by going into the prefab.

Additionally users have the option of enabling auto transactions from the wallets GUI, instead of being prompted it will just go through, it's useful for fast paced games.

![image](https://github.com/ChainSafe/web3.unity/assets/57473220/d4b81f61-71f3-4bee-88b4-cd79f4b6fee5)
![image](https://github.com/ChainSafe/web3.unity/assets/57473220/77a1e800-f120-4ce4-ad56-9ab3b9ad4e64)

To Test:

Run setup.sh(mac) or setup.bat(windows)
Save server settings to populate block explorer field.
Boot up the web3auth login scene.
Enable the wallet in the login logic object.
Test transactions on the sample scene.
Check menus & buttons.

Expected Results:

User is prompted for transactions if specified, otherwise transactions will populate in the tx history area of the prefab.
All login config options are functioning as intended (hide wallet auto tx etc)
Colour, icon & text changes persist.
All buttons and functions are working.
Test editor, standalone & WebGL builds.

Closes #1011
